include/util/viewport.hpp

```diff
@@ -38,7 +38,10 @@ unsigned getFittedZoom(util::Coordinate south_west, util::Coordinate north_east)
     const double height_ratio = (max_y - min_y) / detail::VIEWPORT_HEIGHT;
     const auto zoom = detail::MAX_ZOOM - std::max(std::log(width_ratio), std::log(height_ratio)) * detail::INV_LOG_2;
 
-    return std::max<unsigned>(detail::MIN_ZOOM, zoom);
+    if (std::isfinite(zoom))
```

Comment:
I feel we should catch this earlier in the code above. There needs to be a check if `height_ratio` and `width_ratio` are bigger than 0. In that case `MAX_ZOOM` (18) needs to be returned.